docs/conf.py

```diff
@@ -94,8 +94,8 @@
 # The encoding of source files.
 # source_encoding = 'utf-8-sig'
 
-# The master toctree document.
-master_doc = 'contents'
+# The root document.
+root_doc = 'contents'
```

Comment:
Sphinx renamed `master_doc` to `root_doc` (without deprecation) in 4.0: https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-master_doc

I think relying on Sphinx 4.0.0, from May, is okay. It supports Python 3.6+. I changed `docs/requirements.txt` to match.